src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -293,6 +298,52 @@ private function autowireMethod(\ReflectionMethod $reflectionMethod, array $argu
     }
 
     /**
+     * Autowires getters.
+     *
+     * @param array $overridenGetters
+     * @param array $autowiredMethod
+     *
+     * @return array
+     */
+    private function autowireOverridenGetters(array $overridenGetters, array $autowiredMethod)
+    {
+        foreach ($autowiredMethod as $reflectionMethod) {
+            if (
+                isset($overridenGetters[$reflectionMethod->name]) ||
+                !method_exists($reflectionMethod, 'getReturnType') ||
+                0 !== $reflectionMethod->getNumberOfParameters() ||
+                $reflectionMethod->isFinal() ||
+                $reflectionMethod->returnsReference() ||
+                !($returnType = $reflectionMethod->getReturnType())
+            ) {
+                continue;
+            }
+
+
+            if (null === $this->types) {
+                $this->populateAvailableTypes();
+            }
+
+            $class = $returnType->__toString();
```

Comment:
`$returnType instanceof \ReflectionNamedType ? $returnType->getName() : $returnType->__toString()`
because `__toString` is deprecated since php 7.1